main.rb

```diff
@@ -0,0 +1,205 @@
+module Enumerable
+  def my_each
+    return to_enum :my_each unless block_given?
+
+    iarr = to_a
+    size = iarr.length
+    x = 0
+    until x == size
+      yield(iarr[x])
+      x += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum :my_each_with_index unless block_given?
+
+    iarr = self
```

Comment:
[Optional]


As you have done in `my_each`, let's use `to_a` instead, This helps handle cases where the enumerable is a `Range`.
Currently using a `Range`, `(0..5).my_each_with_index{|a,b| puts "#{a} #{b}"}` gives an error.